### DISCUSSION TITLE
venv/bin/klever-deploy-local: command not found

### DISCUSSION THREAD

**Discussion by wuyuxuan2001** [ldv-klever/klever#8] [Q&A]
I followed document 
"https://klever.readthedocs.io/en/latest/deploy.html"
step by step to deploy Klever on ubuntu 20.04. 
When i am working on "Local Deployment" and run 
"sudo venv/bin/klever-deploy-local --deployment-directory $KLEVER_DEPLOY_DIR install production" 
i failed, and it shows "sudo: venv/bin/klever-deploy-local: command not found".
I am deeply grateful for any assistance you may be able to provide.

**Comment by eunovm** [comment]
Directory "venv" and file "venv/bin/klever-deploy-local" should be created within the root directory of Klever's Git repository, if you accurately do all specified actions and everything goes well. Then you should run the final installation command for local deployment from exactly that directory. Please, ensure that you did do everything necessary successfully. If it will not work still, please, provide a list of commands, that you executed, and names of directories, where you ran these commands.

**Comment by wuyuxuan2001** [comment]
Firstly, I would like to thank you for considering my problem.I tried again, but the same problem occurred.Here is my operation process.
1) git clone --depth 1 --branch v3.7 https://forge.ispras.ru/git/klever.git
After executing command 1), I downloaded the project files.
Then I downloaded the python-debian-11-3.10.2.tar.xz file and executed command 2)
2) sudo tar -xf python-debian-11-3.10.2.tar.xz -C /
3) /usr/local/python3.10-klever/bin/python3 -m venv venv
After command 2) is executed successfully, I run command 3). 
However, because the following error occurs in my pip in the virtual environment and don't know how to fix it,  my subsequent commands are not executed in the virtual environment.
(WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.)
4) pip install -r requirements.txt
5) pip install --upgrade -r requirements.txt
I don't know why, conmands 4) and 5) have the aforementioned WARNING in a virtual environment, but they work fine in a normal environment
Then I created the addons and build bases directories under the klever directory and downloaded all the files following the links given in the klever/klever/depolys/conf/klever.json file.
6) sudo venv/bin/klever-deploy-local --deployment-directory $KLEVER_DEPLOY_DIR install production
Finally, I tried to execute command 6) but found that the klever-deploy-local command did not exist.
Thanks again for your kind help.

**Reply by eunovm** [reply]
It looks like the precompiled Python's distribution does not work in your environment. It is not necessary to use exactly it since it is just standard Python configured somehow and built to deal well with Klever in common environments. I am wondering do you have working Python 3.10+ on your system from any source? For instance, you can try to download and install an appropriate binary package or build it from sources according to numerous instructions on the web. After that you should be able to use it to create the virtual environment and run all subsequent commands without any issues.
